src/components/Output/index.tsx

```diff
@@ -40,29 +41,76 @@ export default class Output extends Component<Props, State> {
     }
   }
 
-  componentDidUpdate(prevProps: Props) {
+  componentDidUpdate(prevProps: Props, prevState: State) {
     if (prevProps.leftImg !== this.props.leftImg && this.canvasLeft) {
       drawBitmapToCanvas(this.canvasLeft, this.props.leftImg);
     }
     if (prevProps.rightImg !== this.props.rightImg && this.canvasRight) {
       drawBitmapToCanvas(this.canvasRight, this.props.rightImg);
     }
+
+    const { scale } = this.state;
+    if (scale !== prevState.scale && this.pinchZoomLeft && this.pinchZoomRight) {
+      // @TODO it would be nice if PinchZoom exposed a variant of setTransform() that
+      // preserved translation. It currently only does this for mouse wheel events.
+      this.pinchZoomLeft.setTransform({ scale });
+      this.pinchZoomRight.setTransform({ scale });
+    }
   }
 
   shouldComponentUpdate(nextProps: Props, nextState: State) {
-    return this.props.leftImg !== nextProps.leftImg ||
-      this.props.rightImg !== nextProps.rightImg ||
-      this.state.verticalTwoUp !== nextState.verticalTwoUp;
+    return !shallowEqual(this.props, nextProps) || !shallowEqual(this.state, nextState);
+  }
+
+  @bind
+  toggleBackground() {
```

Comment:
Throughout this file: we tend to use `onWhatever` for events.

Reply:
Just seems like a less-than-useful nomenclature when there's no `event` argument that couples the method to a UI interaction.

Reply:
Fair enough